### PR TITLE
Bump min python version to 3.10

### PR DESCRIPTION
**Pull request by MridulS**
fixes https://github.com/networkx/nx-guides/issues/138

### CODE REVIEW

File: runtime.txt [hunk 1]
```diff
@@ -1 +1 @@
-python-3.9
+python-3.10
```
Comment (bsipocz) [inline runtime.txt:1]: Could we bump this more? I mean we should test with older version, but binder could run on latest and greatest.

Comment (MridulS) [reply runtime.txt:1]: I think we wanted to keep this on the min supported version so all the deps also work fine? I don't exactly remember why we don't use the latest python version, maybe @rossbar ?

Comment (bsipocz) [reply runtime.txt:1]: Yes, but CI can be used to ensure that, I mean there should be a newer combination where all dependencies work.

**Review by bsipocz — APPROVED** [review]
Not that it matters here 😄

**Pull request merged**